libbeat/beat/beat.go

```diff
@@ -469,8 +469,30 @@ func (b *Beat) registerTemplateLoading() error {
 		}
 	}
 
-	esConfig := b.Config.Output["elasticsearch"]
+	// Check if outputting to file is enabled, and output to file if it is
+	if b.Config.Template != nil && b.Config.Template.Enabled() {
+		var cfg template.TemplateConfig
+		err := b.Config.Template.Unpack(&cfg)
+		if err != nil {
+			return fmt.Errorf("unpacking template config fails: %v", err)
+		}
+		if len(cfg.OutputToFile.Path) > 0 {
+			// output to file is enabled
+			loader, err := template.NewLoader(b.Config.Template, nil, b.Info)
+			if err != nil {
+				return fmt.Errorf("Error creating Elasticsearch template loader: %v", err)
+			}
+			err = loader.Generate()
+			if err != nil {
+				return fmt.Errorf("Error generating template: %v", err)
+			}
 
+			// XXX: Should we kill the Beat here or just continue?
+			return fmt.Errorf("Stopping after successfully writing the template to the file.")
```

Comment:
In this case, dear reviewdog, we know that the error is final, so the punctuation looks better to the user.